modules/sdk/src/main/scala/org/tessellation/sdk/domain/statechannel/StateChannelValidator.scala

```diff
@@ -40,12 +44,26 @@ object StateChannelValidator {
       signedValidator
         .isSignedExclusivelyBy(signedSC, address)
         .map(_.errorMap[StateChannelValidationError](_ => NotSignedExclusivelyByStateChannelOwner))
+
+    private def validateSnapshotSize(
+      signedSC: Signed[StateChannelSnapshotBinary]
+    ): F[StateChannelValidationErrorOr[Signed[StateChannelSnapshotBinary]]] =
+      signedSC.toBinaryF.map { binary =>
+        val actualSize = binary.size
+        val isWithinLimit = actualSize <= maxBinarySizeInBytes
+
+        if (isWithinLimit)
+          signedSC.validNec
+        else
+          BinaryExceedsMaxAllowedSize(maxBinarySizeInBytes, actualSize).invalidNec
+      }
   }
 
   @derive(eqv, show)
   sealed trait StateChannelValidationError
   case class InvalidSigned(error: SignedValidationError) extends StateChannelValidationError
   case object NotSignedExclusivelyByStateChannelOwner extends StateChannelValidationError
+  case class BinaryExceedsMaxAllowedSize(maxSize: Long, was: Int) extends StateChannelValidationError
```

Comment:
Probably it's better to use the same types for both cases. I guess that would be `Int` or `NonNegInt`.

Reply:
Size for an array returns `Int`, so I went with `Int` to not upcast it to long. And wanted to use long for the maxBytesSize to be safe.